x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/X-Block.java.st

```diff
@@ -54,6 +54,35 @@ $else$
     $type$ get$Type$(int valueIndex);
 $endif$
 
+    /**
+     * Checks if this block has the given value at valueIndex. If at this index we have a
+     * multivalue, then it returns true if any values match.
+     *
+     * @param valueIndex the index at which we should check the value(s)
+     * @param value the value to check against
+     */
+    default boolean hasValue(int valueIndex, $type$ value) {
+        final var count = getValueCount(valueIndex);
+        final var startIndex = getFirstValueIndex(valueIndex);
+        for (int index = startIndex; index < startIndex + count; index++) {
+$if(BytesRef)$
+            var ref = new BytesRef();
```

Comment:
This scratch creation can be moved outside, so we avoid creating it on every iteration

Reply:
doh, yeah that was the reason to have the param in the first place. Moved it out of the loop in both places.